pinot-common/src/main/java/org/apache/pinot/common/data/FieldSpec.java

```diff
@@ -153,6 +154,11 @@ public void setVirtualColumnProvider(String virtualColumnProvider) {
     _virtualColumnProvider = virtualColumnProvider;
   }
 
+  @JsonIgnore
+  public boolean isVirtualColumn() {
+    return _virtualColumnProvider != null && !_virtualColumnProvider.isEmpty();
```

Comment:
Are we allowing column names to start with $ for physical columns as well? Or, are we allowing virtual  column names to start with characters other than "$"? I did not see checks for either of these (so happens that the in0built virtual columns are starting with dollar sign). Would be good to add a comment here about that.

Reply:
Virtual column works only when the provider is provided. Added comments about that